File: client/src/models/Audit.ts

export interface Audit {
    id: Number;
    title: string;
    message: string;
    timeStamp: Number;
}

export class UserAudit implements Audit {
    user: string;
    ip: string;
    id: Number;
    message: string;
    timeStamp: number;
    title: string;

    constructor(id: Number, title: string, message: string, timeStamp: number, user: string, ip: string) {
        this.id = id;
        this.title = title;
        this.message = message;
        this.timeStamp = timeStamp;
        this.user = user;
        this.ip = ip;
    }
}

export class ApiAudit {
    api: string;
    id: Number;
    message: string;
    timeStamp: number;
    title: string;

    constructor(id: Number, title: string, message: string, timeStamp: number, api: string) {
        this.id = id;
        this.title = title;
        this.message = message;
        this.timeStamp = timeStamp;
        this.api = api
    }
}

File: src/main/java/com/rsj/aerion/config/models/Audit.java
package com.rsj.aerion.config.models;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

@MappedSuperclass
public class Audit {

    Audit() {

    }

    public Audit(String title, String message, Long timeStamp) {
        this.title = title;
        this.message = message;
        this.timeStamp = timeStamp;
    }

    public Audit(Long id, String title, String message, Long timeStamp) {
        this.id = id;
        this.title = title;
        this.message = message;
        this.timeStamp = timeStamp;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String title;
    private String message;
    private Long timeStamp;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Long getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(Long timeStamp) {
        this.timeStamp = timeStamp;
    }
}
